src/execution_impl.rs

```diff
@@ -64,23 +63,23 @@ fn how_many(env: &Env) -> i32 {
 /// An utility print function to write on stdout directly from AssemblyScript:
 fn assembly_script_print(env: &Env, arg: i32) {
     let str_ptr = StringPtr::new(arg as u32);
-    let print: fn(&str) -> Result<()> = env.interface.print;
-    print(
-        &str_ptr
-            .read(env.wasm_env.memory.get_ref().expect("uninitialized memory"))
-            .unwrap(),
-    )
-    .unwrap();
+    env.interface
+        .print(
+            &str_ptr
+                .read(env.wasm_env.memory.get_ref().expect("uninitialized memory"))
+                .unwrap(),
+        )
+        .unwrap();
 }
 
 /// Create an instance of VM from a module with a given interface, an operation
 /// number limit and a webassembly module
-fn create_instance(limit: u64, module: &[u8], interface: &Interface) -> Result<Instance> {
+fn create_instance(limit: u64, module: &[u8], interface: &dyn Interface) -> Result<Instance> {
     let metering = Arc::new(Metering::new(limit, |_: &Operator| -> u64 { 1 }));
     let mut compiler_config = Cranelift::default();
     compiler_config.push_middleware(metering);
     let store = Store::new(&Universal::new(compiler_config).engine());
-    let env = Env::new(interface);
+    let env = Env::new(&*interface);
```

Comment:
I don't understand why a `&*` is needed, shouldn't that be equivalent to just nothing ?

Reply:
No its not

Reply:
I think the `*` gets you "what is inside the box", and then you pass it by reference. 

Actually it was unnecessary here because we're already getting a ref as argument, but it seems necessary elsewhere like in the current tests.

Reply:
Thanks !